pkg/delete/internal/classic/delete.go

```diff
@@ -79,29 +89,33 @@ func is404(err error) bool {
 }
 
 // resolveIdentifier get the actual ID from DT and update entries with it
-func resolveIdentifier(ctx context.Context, client dtclient.Client, theAPI *api.API, identifier string) (string, error) {
+func resolveIdentifier(ctx context.Context, client dtclient.Client, theAPI *api.API, identifier identifier) (string, error) {
 	knownValues, err := client.ListConfigs(ctx, *theAPI)
 	if err != nil {
 		return "", err
 	}
 
-	id, err := findUniqueID(knownValues, identifier)
+	id, err := findUniqueID(knownValues, identifier, theAPI.CheckEqualFunc)
 	if err != nil {
 		return "", err
 	}
 
 	return id, nil
 }
 
-func findUniqueID(knownValues []dtclient.Value, identifier string) (string, error) {
+func findUniqueID(knownValues []dtclient.Value, identifier identifier, uniqeFn func(map[string]any, map[string]any) bool) (string, error) {
```

Comment:
nit: maybe rename `uniqeFn`to `uniqueFn`

Reply:
Although, if this is really just the `api.CheckEqualFunc`, we should use that as a name, or come up with a better more general name and use it both places.